Test Failure: ClusterServiceTests. testClusterStateUpdateLogging

Fails periodically but rarely (1-2 times a month) on 5.1, 5.2, 5.3, 5.x and master branches. 

```
Suite: org.elasticsearch.cluster.service.ClusterServiceTests
  1> [2017-03-24T10:42:28,985][INFO ][o.e.c.s.ClusterServiceTests] [testSingleBatchSubmission]: before test
  1> [2017-03-24T10:42:28,992][INFO ][o.e.c.s.ClusterServiceTests$TimedClusterService] new_master {node1}{Us26Ll-DRY6tGqIwXRgx8g}{local}{local[295]}, reason: test setting state
  1> [2017-03-24T10:42:28,995][INFO ][o.e.c.s.ClusterServiceTests] [testSingleBatchSubmission]: after test
  1> [2017-03-24T10:42:28,997][INFO ][o.e.c.s.ClusterServiceTests] [testClusterStateBatchedUpdates]: before test
  1> [2017-03-24T10:42:29,000][INFO ][o.e.c.s.ClusterServiceTests$TimedClusterService] new_master {node1}{VvwSV_nRSMuBLNPEogvXuQ}{local}{local[296]}, reason: test setting state
  1> [2017-03-24T10:42:29,160][INFO ][o.e.c.s.ClusterServiceTests] [testClusterStateBatchedUpdates]: after test
  1> [2017-03-24T10:42:29,164][INFO ][o.e.c.s.ClusterServiceTests] [testClusterStateUpdateLogging]: before test
  1> [2017-03-24T10:42:29,168][TRACE][o.e.c.s.ClusterServiceTests$TimedClusterService] will process [test setting state]
  1> [2017-03-24T10:42:29,168][DEBUG][o.e.c.s.ClusterServiceTests$TimedClusterService] processing [test setting state]: execute
  1> [2017-03-24T10:42:29,169][TRACE][o.e.c.s.ClusterServiceTests$TimedClusterService] cluster state updated, source [test setting state]
  1> cluster uuid: _na_
  1> version: 2
  1> state uuid: 31YhffagTya2Wnf2FvVoXw
  1> from_diff: false
  1> meta data version: 0
  1> nodes: 
  1>    {node1}{gJfhYupGSQWSvbSpf5bxUw}{local}{local[297]}, local, master
  1> routing_table (version 0):
  1> routing_nodes:
  1> ---- unassigned
  1> [2017-03-24T10:42:29,170][INFO ][o.e.c.s.ClusterServiceTests$TimedClusterService] new_master {node1}{gJfhYupGSQWSvbSpf5bxUw}{local}{local[297]}, reason: test setting state
  1> [2017-03-24T10:42:29,170][DEBUG][o.e.c.s.ClusterServiceTests$TimedClusterService] publishing cluster state version [2]
  1> [2017-03-24T10:42:29,172][DEBUG][o.e.c.s.ClusterServiceTests$TimedClusterService] applying cluster state version 2
  1> [2017-03-24T10:42:29,172][DEBUG][o.e.c.s.ClusterServiceTests$TimedClusterService] set local cluster state to version 2
  1> [2017-03-24T10:42:29,172][TRACE][o.e.c.s.ClusterServiceTests$TimedClusterService] calling [org.elasticsearch.cluster.service.ClusterService$LocalNodeMasterListeners@39ceead6] with change to version [2]
  1> [2017-03-24T10:42:29,172][DEBUG][o.e.c.s.ClusterServiceTests$TimedClusterService] processing [test setting state]: took [3ms] done applying updated cluster_state (version: 2, uuid: 31YhffagTya2Wnf2FvVoXw)
  1> [2017-03-24T10:42:29,179][TRACE][o.e.c.s.ClusterServiceTests$TimedClusterService] will process [test1]
  1> [2017-03-24T10:42:29,179][DEBUG][o.e.c.s.ClusterServiceTests$TimedClusterService] processing [test1]: execute
  1> [2017-03-24T10:42:29,180][DEBUG][o.e.c.s.ClusterServiceTests$TimedClusterService] processing [test1]: took [1s] no change in cluster_state
  1> [2017-03-24T10:42:29,181][TRACE][o.e.c.s.ClusterServiceTests$TimedClusterService] will process [test2]
  1> [2017-03-24T10:42:29,181][DEBUG][o.e.c.s.ClusterServiceTests$TimedClusterService] processing [test2]: execute
  1> [2017-03-24T10:42:29,182][TRACE][o.e.c.s.ClusterServiceTests$TimedClusterService] failed to execute cluster state update in [2s], state:
  1> version [2], source [test2]
  1> nodes: 
  1>    {node1}{gJfhYupGSQWSvbSpf5bxUw}{local}{local[297]}, local, master
  1> routing_table (version 0):
  1> routing_nodes:
  1> ---- unassigned
  1> java.lang.IllegalArgumentException: Testing handling of exceptions in the cluster state task
  1> 	at org.elasticsearch.cluster.service.ClusterServiceTests$18.execute(ClusterServiceTests.java:867) ~[test/:?]
  1> 	at org.elasticsearch.cluster.ClusterStateUpdateTask.execute(ClusterStateUpdateTask.java:45) ~[main/:5.4.0-SNAPSHOT]
  1> 	at org.elasticsearch.cluster.service.ClusterService.executeTasks(ClusterService.java:679) ~[main/:?]
  1> 	at org.elasticsearch.cluster.service.ClusterService.calculateTaskOutputs(ClusterService.java:658) ~[main/:?]
  1> 	at org.elasticsearch.cluster.service.ClusterService.runTasks(ClusterService.java:617) ~[main/:?]
  1> 	at org.elasticsearch.cluster.service.ClusterService$UpdateTask.run(ClusterService.java:1117) ~[main/:?]
  1> 	at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:544) ~[main/:?]
  1> 	at org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.runAndClean(PrioritizedEsThreadPoolExecutor.java:238) ~[main/:?]
  1> 	at org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.run(PrioritizedEsThreadPoolExecutor.java:201) ~[main/:?]
  1> 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) ~[?:1.8.0_111]
  1> 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) ~[?:1.8.0_111]
  1> 	at java.lang.Thread.run(Thread.java:745) ~[?:1.8.0_111]
  1> [2017-03-24T10:42:29,184][DEBUG][o.e.c.s.ClusterServiceTests$TimedClusterService] processing [test2]: took [2s] no change in cluster_state
  1> [2017-03-24T10:42:29,184][TRACE][o.e.c.s.ClusterServiceTests$TimedClusterService] will process [test3]
  1> [2017-03-24T10:42:29,184][DEBUG][o.e.c.s.ClusterServiceTests$TimedClusterService] processing [test3]: execute
  1> [2017-03-24T10:42:29,184][TRACE][o.e.c.s.ClusterServiceTests$TimedClusterService] cluster state updated, source [test3]
  2> REPRODUCE WITH: gradle :core:test -Dtests.seed=4750D393F4387E4B -Dtests.class=org.elasticsearch.cluster.service.ClusterServiceTests -Dtests.method="testClusterStateUpdateLogging" -Dtests.security.manager=true -Dtests.locale=en-PH -Dtests.timezone=Zulu
  1> cluster uuid: _na_
  1> version: 4
  1> state uuid: qoJTIhLARm22d_Wb1_-fIw
  1> from_diff: false
  1> meta data version: 0
  1> nodes: 
  1>    {node1}{gJfhYupGSQWSvbSpf5bxUw}{local}{local[297]}, local, master
  1> routing_table (version 0):
  1> routing_nodes:
  1> ---- unassigned
  1> [2017-03-24T10:42:29,184][DEBUG][o.e.c.s.ClusterServiceTests$TimedClusterService] publishing cluster state version [4]
  1> [2017-03-24T10:42:29,184][DEBUG][o.e.c.s.ClusterServiceTests$TimedClusterService] applying cluster state version 4
  1> [2017-03-24T10:42:29,184][DEBUG][o.e.c.s.ClusterServiceTests$TimedClusterService] set local cluster state to version 4
  1> [2017-03-24T10:42:29,184][TRACE][o.e.c.s.ClusterServiceTests$TimedClusterService] calling [org.elasticsearch.cluster.service.ClusterService$LocalNodeMasterListeners@39ceead6] with change to version [4]
  1> [2017-03-24T10:42:29,184][DEBUG][o.e.c.s.ClusterServiceTests$TimedClusterService] processing [test3]: took [3s] done applying updated cluster_state (version: 4, uuid: qoJTIhLARm22d_Wb1_-fIw)
  1> [2017-03-24T10:42:29,184][TRACE][o.e.c.s.ClusterServiceTests$TimedClusterService] will process [test4]
  1> [2017-03-24T10:42:29,185][DEBUG][o.e.c.s.ClusterServiceTests$TimedClusterService] processing [test4]: execute
  1> 2017-03-24 10:42:29,186 elasticsearch[clusterService#updateTask][T#1] ERROR Attempted to append to non-started appender mock
  1> [2017-03-24T10:42:29,185][DEBUG][o.e.c.s.ClusterServiceTests$TimedClusterService] processing [test4]: took [0s] no change in cluster_state
FAILURE 0.04s J0 | ClusterServiceTests.testClusterStateUpdateLogging <<< FAILURES!
   > Throwable #1: java.lang.AssertionError: 
   > Expected: an empty collection
   >      but: <[Attempted to append to non-started appender mock]>
   > 	at __randomizedtesting.SeedInfo.seed([4750D393F4387E4B:EEA6460CBA279397]:0)
   > 	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
   > 	at org.elasticsearch.test.ESTestCase.checkStaticState(ESTestCase.java:379)
   > 	at org.elasticsearch.test.ESTestCase.after(ESTestCase.java:262)
   > 	at java.lang.Thread.run(Thread.java:745)
  1> [2017-03-24T10:42:29,199][INFO ][o.e.c.s.ClusterServiceTests] [testLocalNodeMasterListenerCallbacks]: before test
  1> [2017-03-24T10:42:29,201][INFO ][o.e.c.s.ClusterServiceTests$TimedClusterService] new_master {node1}{GBu-l3igQRqd_40IMx6Qmg}{local}{local[298]}, reason: test setting state
  1> [2017-03-24T10:42:29,203][INFO ][o.e.c.s.ClusterServiceTests$TimedClusterService] new_master {node1}{0ss1FrODSAqQUvCA5ffcJA}{local}{local[299]}, reason: test setting state
  1> [2017-03-24T10:42:29,204][INFO ][o.e.c.s.ClusterServiceTests$TimedClusterService] new_master {node1}{0ss1FrODSAqQUvCA5ffcJA}{local}{local[299]}, reason: test setting state
  1> [2017-03-24T10:42:29,204][INFO ][o.e.c.s.ClusterServiceTests] [testLocalNodeMasterListenerCallbacks]: after test
  1> [2017-03-24T10:42:29,206][INFO ][o.e.c.s.ClusterServiceTests] [testClusterStateUpdateTasksAreExecutedInOrder]: before test
  1> [2017-03-24T10:42:29,208][INFO ][o.e.c.s.ClusterServiceTests$TimedClusterService] new_master {node1}{JZU5dkNnQwik-8ul80jtCw}{local}{local[300]}, reason: test setting state
  1> [2017-03-24T10:42:29,387][INFO ][o.e.c.s.ClusterServiceTests] [testClusterStateUpdateTasksAreExecutedInOrder]: after test
  1> [2017-03-24T10:42:29,389][INFO ][o.e.c.s.ClusterServiceTests] [testPrioritizedTasks]: before test
  1> [2017-03-24T10:42:29,394][INFO ][o.e.c.s.ClusterServiceTests$TimedClusterService] new_master {node1}{UTcHwFD4S3OqVJOorAwNbA}{local}{local[301]}, reason: test setting state
  1> [2017-03-24T10:42:29,396][INFO ][o.e.c.s.ClusterServiceTests] [testPrioritizedTasks]: after test
  1> [2017-03-24T10:42:29,398][INFO ][o.e.c.s.ClusterServiceTests] [testClusterStateApplierCantSampleClusterState]: before test
  1> [2017-03-24T10:42:29,405][INFO ][o.e.c.s.ClusterServiceTests$TimedClusterService] new_master {node1}{6WnJWJwXTsuC5pJFoS26Kw}{local}{local[302]}, reason: test setting state
  1> [2017-03-24T10:42:29,406][INFO ][o.e.c.s.ClusterServiceTests] [testClusterStateApplierCantSampleClusterState]: after test
  1> [2017-03-24T10:42:29,408][INFO ][o.e.c.s.ClusterServiceTests] [testMasterAwareExecution]: before test
  1> [2017-03-24T10:42:29,411][INFO ][o.e.c.s.ClusterServiceTests$TimedClusterService] new_master {node1}{z54dh55bTq2LyIEmE-6YNw}{local}{local[303]}, reason: test setting state
  1> [2017-03-24T10:42:29,417][INFO ][o.e.c.s.ClusterServiceTests] [testMasterAwareExecution]: after test
  1> [2017-03-24T10:42:29,424][WARN ][o.e.c.s.ClusterServiceTests$TimedClusterService] cluster state update task [test2] took [32s] above the warn threshold of 30s
  1> [2017-03-24T10:42:29,424][WARN ][o.e.c.s.ClusterServiceTests$TimedClusterService] cluster state update task [test2] took [32s] above the warn threshold of 30s
  1> [2017-03-24T10:42:29,424][WARN ][o.e.c.s.ClusterServiceTests$TimedClusterService] cluster state update task [test3] took [33s] above the warn threshold of 30s
  1> [2017-03-24T10:42:29,425][WARN ][o.e.c.s.ClusterServiceTests$TimedClusterService] cluster state update task [test4] took [34s] above the warn threshold of 30s
  1> [2017-03-24T10:42:29,431][INFO ][o.e.c.s.ClusterServiceTests] [testClusterStateTaskListenerThrowingExceptionIsOkay]: before test
  1> [2017-03-24T10:42:29,440][INFO ][o.e.c.s.ClusterServiceTests$TimedClusterService] new_master {node1}{LPB1z4FVTE-mN6y_OcfLYw}{local}{local[306]}, reason: test setting state
  1> [2017-03-24T10:42:29,444][ERROR][o.e.c.s.ClusterServiceTests$TimedClusterService] exception thrown by listener while notifying of cluster state processed from [testClusterStateTaskListenerThrowingExceptionIsOkay], old cluster state:
  1> cluster uuid: _na_
  1> version: 2
  1> state uuid: I7gZBOsGSgOjW7ho9hlyLA
  1> from_diff: false
  1> meta data version: 0
  1> nodes: 
  1>    {node1}{LPB1z4FVTE-mN6y_OcfLYw}{local}{local[306]}, local, master
  1> routing_table (version 0):
  1> routing_nodes:
  1> ---- unassigned
  1> new cluster state:
  1> cluster uuid: _na_
  1> version: 3
  1> state uuid: f_6etjQ8TMuwbgAER7zbHA
  1> from_diff: false
  1> meta data version: 0
  1> nodes: 
  1>    {node1}{LPB1z4FVTE-mN6y_OcfLYw}{local}{local[306]}, local, master
  1> routing_table (version 0):
  1> routing_nodes:
  1> ---- unassigned
  1> java.lang.IllegalStateException: testClusterStateTaskListenerThrowingExceptionIsOkay
  1> 	at org.elasticsearch.cluster.service.ClusterServiceTests$10.clusterStateProcessed(ClusterServiceTests.java:339) ~[test/:?]
  1> 	at org.elasticsearch.cluster.service.ClusterService$SafeClusterStateTaskListener.clusterStateProcessed(ClusterService.java:1022) ~[main/:?]
  1> 	at org.elasticsearch.cluster.service.ClusterService$TaskOutputs.lambda$processedDifferentClusterState$1(ClusterService.java:916) ~[main/:?]
  1> 	at java.util.ArrayList.forEach(ArrayList.java:1249) ~[?:1.8.0_111]
  1> 	at org.elasticsearch.cluster.service.ClusterService$TaskOutputs.processedDifferentClusterState(ClusterService.java:916) ~[main/:?]
  1> 	at org.elasticsearch.cluster.service.ClusterService.publishAndApplyChanges(ClusterService.java:842) ~[main/:?]
  1> 	at org.elasticsearch.cluster.service.ClusterService.runTasks(ClusterService.java:633) ~[main/:?]
  1> 	at org.elasticsearch.cluster.service.ClusterService$UpdateTask.run(ClusterService.java:1117) ~[main/:?]
  1> 	at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:544) ~[main/:?]
  1> 	at org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.runAndClean(PrioritizedEsThreadPoolExecutor.java:238) ~[main/:?]
  1> 	at org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.run(PrioritizedEsThreadPoolExecutor.java:201) ~[main/:?]
  1> 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) ~[?:1.8.0_111]
  1> 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) ~[?:1.8.0_111]
  1> 	at java.lang.Thread.run(Thread.java:745) ~[?:1.8.0_111]
  1> [2017-03-24T10:42:29,445][INFO ][o.e.c.s.ClusterServiceTests] [testClusterStateTaskListenerThrowingExceptionIsOkay]: after test
  1> [2017-03-24T10:42:29,448][INFO ][o.e.c.s.ClusterServiceTests] [testBlockingCallInClusterStateTaskListenerFails]: before test
  1> [2017-03-24T10:42:29,451][INFO ][o.e.c.s.ClusterServiceTests$TimedClusterService] new_master {node1}{x7DCbkb_Sv2GgAjPUDkQNw}{local}{local[307]}, reason: test setting state
  1> [2017-03-24T10:42:29,453][INFO ][o.e.c.s.ClusterServiceTests] [testBlockingCallInClusterStateTaskListenerFails]: after test
  1> [2017-03-24T10:42:29,455][INFO ][o.e.c.s.ClusterServiceTests] [testDisconnectFromNewlyAddedNodesIfClusterStatePublishingFails]: before test
  1> [2017-03-24T10:42:29,459][INFO ][o.e.c.s.ClusterServiceTests$TimedClusterService] new_master {node1}{Bn8RrsDNQoSEvy4JLd7P6Q}{local}{local[308]}, reason: test setting state
  1> [2017-03-24T10:42:29,473][INFO ][o.e.c.s.ClusterServiceTests$TimedClusterService] new_master {node1}{M2ePtn7FTnCBSzAEzkYKsQ}{local}{local[309]}, reason: test setting state
  1> [2017-03-24T10:42:29,477][INFO ][o.e.c.s.ClusterServiceTests$TimedClusterService] added {{node2}{SxyuFjitREK0DpSzuei2dQ}{local}{local[310]},}, reason: test
  1> [2017-03-24T10:42:29,478][WARN ][o.e.c.s.ClusterServiceTests$TimedClusterService] failing [test]: failed to commit cluster state version [3]
  1> org.elasticsearch.discovery.Discovery$FailedToCommitClusterStateException: just to test this
  1> 	at org.elasticsearch.cluster.service.ClusterServiceTests.lambda$testDisconnectFromNewlyAddedNodesIfClusterStatePublishingFails$12(ClusterServiceTests.java:1076) ~[test/:?]
  1> 	at org.elasticsearch.cluster.service.ClusterService.publishAndApplyChanges(ClusterService.java:788) ~[main/:?]
  1> 	at org.elasticsearch.cluster.service.ClusterService.runTasks(ClusterService.java:633) ~[main/:?]
  1> 	at org.elasticsearch.cluster.service.ClusterService$UpdateTask.run(ClusterService.java:1117) ~[main/:?]
  1> 	at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:544) ~[main/:?]
  1> 	at org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.runAndClean(PrioritizedEsThreadPoolExecutor.java:238) ~[main/:?]
  1> 	at org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.run(PrioritizedEsThreadPoolExecutor.java:201) ~[main/:?]
  1> 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) ~[?:1.8.0_111]
  1> 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) ~[?:1.8.0_111]
  1> 	at java.lang.Thread.run(Thread.java:745) ~[?:1.8.0_111]
  1> [2017-03-24T10:42:29,481][INFO ][o.e.c.s.ClusterServiceTests] [testDisconnectFromNewlyAddedNodesIfClusterStatePublishingFails]: after test
  1> [2017-03-24T10:42:29,483][INFO ][o.e.c.s.ClusterServiceTests] [testTimeoutUpdateTask]: before test
  1> [2017-03-24T10:42:29,493][INFO ][o.e.c.s.ClusterServiceTests$TimedClusterService] new_master {node1}{PPfuJQSxT4CkA8iLTM68iQ}{local}{local[311]}, reason: test setting state
  1> [2017-03-24T10:42:29,499][INFO ][o.e.c.s.ClusterServiceTests] [testTimeoutUpdateTask]: after test
  1> [2017-03-24T10:42:29,500][INFO ][o.e.c.s.ClusterServiceTests] [testOneExecutorDontStarveAnother]: before test
  1> [2017-03-24T10:42:29,502][INFO ][o.e.c.s.ClusterServiceTests$TimedClusterService] new_master {node1}{UP3JsEP3Req6B0vxzCFQeg}{local}{local[312]}, reason: test setting state
  1> [2017-03-24T10:42:29,505][INFO ][o.e.c.s.ClusterServiceTests] [testOneExecutorDontStarveAnother]: after test
  1> [2017-03-24T10:42:29,507][INFO ][o.e.c.s.ClusterServiceTests] [testDuplicateSubmission]: before test
  1> [2017-03-24T10:42:29,522][INFO ][o.e.c.s.ClusterServiceTests$TimedClusterService] new_master {node1}{a6HtYxkASg6GpfUrgVMRPA}{local}{local[313]}, reason: test setting state
  1> [2017-03-24T10:42:29,527][INFO ][o.e.c.s.ClusterServiceTests] [testDuplicateSubmission]: after test
  1> [2017-03-24T10:42:29,539][INFO ][o.e.c.s.ClusterServiceTests] [testTimedOutUpdateTaskCleanedUp]: before test
  1> [2017-03-24T10:42:29,543][INFO ][o.e.c.s.ClusterServiceTests$TimedClusterService] new_master {node1}{l4dyH25TR3ebpfTUBcABRw}{local}{local[314]}, reason: test setting state
  1> [2017-03-24T10:42:29,546][INFO ][o.e.c.s.ClusterServiceTests] [testTimedOutUpdateTaskCleanedUp]: after test
  1> [2017-03-24T10:42:29,550][INFO ][o.e.c.s.ClusterServiceTests] [testClusterStateApplierCanCreateAnObserver]: before test
  1> [2017-03-24T10:42:29,555][INFO ][o.e.c.s.ClusterServiceTests$TimedClusterService] new_master {node1}{C1T8a4alTLO9HxHFNBLiKg}{local}{local[315]}, reason: test setting state
  1> [2017-03-24T10:42:29,558][INFO ][o.e.c.s.ClusterServiceTests] [testClusterStateApplierCanCreateAnObserver]: after test
  2> NOTE: leaving temporary files on disk at: /var/lib/jenkins/workspace/elastic+elasticsearch+5.x+multijob-unix-compatibility/os/opensuse/core/build/testrun/test/J0/temp/org.elasticsearch.cluster.service.ClusterServiceTests_4750D393F4387E4B-001
  2> NOTE: test params are: codec=Asserting(Lucene62): {}, docValues:{}, maxPointsInLeafNode=344, maxMBSortInHeap=5.831578072249608, sim=RandomSimilarity(queryNorm=true,coord=no): {}, locale=en-PH, timezone=Zulu
  2> NOTE: Linux 3.16.7-53-default amd64/Oracle Corporation 1.8.0_111 (64-bit)/cpus=4,threads=1,free=395147848,total=530579456
  2> verTests, ScheduleWithFixedDelayTests, OsStatsTests, CountedCollectorTests, FilterAllocationDeciderTests, MatchPhrasePrefixQueryBuilderTests, PipelineExecutionServiceTests, JvmMonitorTests, CompletionSuggesterBuilderTests, TrackingResultProcessorTests, LegacyTokenCountFieldMapperTests, NetworkServiceTests, TaskTests, CustomPassageFormatterTests, CharMatcherTests, ConstructingObjectParserTests, DiscoveryNodesTests, SpatialStrategyTests, RestIndicesActionTests, MoreLikeThisQueryBuilderTests, URIPatternTests, SnapshotUtilsTests, GeoContextMappingTests, MultiOrdinalsTests, IndexTests, DateFieldTypeTests, UpdateNumberOfReplicasTests, MetaDataDeleteIndexServiceTests, IdFieldMapperTests, PercentileRanksTests, ShardStateActionTests, BytesStreamsTests, DecisionsImpactOnClusterHealthTests, BlendedTermQueryTests, GeoJSONShapeParserTests, AllocationDecisionTests, ClusterRebalanceRoutingTests, InnerHitBuilderTests, SettingsTests, CommandTests, BinaryFieldMapperTests, GroupShardsIteratorTests, SimulatePipelineRequestTests, ClusterAllocationExplainActionTests, AvgTests, DateMathParserTests, IndexIdTests, FileInfoTests, IndexRequestTests, NullValueObjectMappingTests, BooleansTests, CopyToMapperTests, CombineFunctionTests, ConcurrentRecyclerTests, CommonGramsTokenFilterFactoryTests, FileWatcherTests, LegacyByteFieldTypeTests, LegacyDoubleFieldTypeTests, ShardsLimitAllocationTests, MinBucketTests, DateMathExpressionResolverTests, GeoDistanceQueryBuilderTests, SizeValueTests, DateHistogramTests, CborXContentTests, IndexShardOperationsLockTests, ASCIIFoldingTokenFilterFactoryTests, LegacyIntegerFieldTypeTests, IndexMetaDataTests, CompletionSuggestionOptionTests, ConstantScoreQueryBuilderTests, TransportNodesActionTests, GetFieldMappingsResponseTests, StartRecoveryRequestTests, TermSuggestionOptionTests, RangeFieldMapperTests, SpanTermQueryBuilderTests, ExpectedShardSizeAllocationTests, MaxTests, AdapterActionFutureTests, FreqTermsEnumTests, IndicesServiceTests, EvictingQueueTests, TableTests, RangeQueryRewriteTests, FieldMaskingSpanQueryBuilderTests, ConfigurationUtilsTests, StoreRecoveryTests, SettingTests, CodecTests, JavaMultiFieldMergeTests, PreferPrimaryAllocationTests, MinScoreScorerTests, ExplainRequestTests, ShardCoreKeyMapTests, FieldStatsProviderRefreshTests, PolygonBuilderTests, StatsTests, NumbersTests, IndicesStatsTests, DerivativeTests, BucketScriptTests, RandomAllocationDeciderTests, SnowballAnalyzerTests, HppcMapsTestsompletionSuggestionOptionTests, ConstantScoreQueryBuilderTests, TransportNodesActionTests, GetFieldMappingsResponseTests, StartRecoveryRequestTests, TermSuggestionOptionTests, RangeFieldMapperTests, SpanTermQueryBuilderTests, ExpectedShardSizeAllocationTests, MaxTests, AdapterActionFutureTests, FreqTermsEnumTests, IndicesServiceTests, EvictingQueueTests, TableTests, RangeQueryRewriteTests, FieldMaskingSpanQueryBuilderTests, ConfigurationUtilsTests, StoreRecoveryTests, SettingTests, CodecTests, JavaMultiFieldMergeTests, PreferPrimaryAllocationTests, MinScoreScorerTests, ExplainRequestTests, ShardCoreKeyMapTests, FieldStatsProviderRefreshTests, PolygonBuilderTests, StatsTests, NumbersTests, IndicesStatsTests, DerivativeTests, BucketScriptTests, RandomAllocationDeciderTests, SnowballAnalyzerTests, HppcMapsTests, UidFieldTypeTests, AvgBucketTests, RecoverySourceHandlerTests, SimulateDocumentSimpleResultTests, ScriptQueryBuilderTests, MultiFieldIncludeInAllMapperTests, NestedTests, HasParentQueryBuilderTests, CompletionsStatsTests, ParentTaskAssigningClientTests, RestControllerTests, GeoPointParsingTests, FsDirectoryServiceTests, DeleteByQueryRequestTests, DirectCandidateGeneratorTests, DynamicMappingDisabledTests, FsBlobStoreContainerTests, GatewayServiceTests, ClusterRerouteTests, MultiGetRequestTests, RoutingFieldMapperTests, AnalysisModuleTests, IngestServiceTests, TransportBulkActionTookTests, FieldNamesFieldMapperTests, IndexBalanceTests, SpanContainingQueryBuilderTests, ClusterServiceTests]
Completed [315/923] on J0 in 0.61s, 17 tests, 1 failure <<< FAILURES!
```

Another: https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+5.6+oracle-periodic/642/console

```
04:55:10    > Throwable #1: java.lang.AssertionError: 
04:55:10    > Expected: an empty collection
04:55:10    >      but: <[Attempted to append to non-started appender mock]>
04:55:10    > 	at __randomizedtesting.SeedInfo.seed([B9C15C2426A1FB86:1037C9BB68BE165A]:0)
04:55:10    > 	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
04:55:10    > 	at org.elasticsearch.test.ESTestCase.checkStaticState(ESTestCase.java:386)
04:55:10    > 	at org.elasticsearch.test.ESTestCase.after(ESTestCase.java:268)
04:55:10    > 	at java.lang.Thread.run(Thread.java:748)
```

Another one in https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+6.4+matrix-java-periodic/ES_BUILD_JAVA=java10,ES_RUNTIME_JAVA=java8,nodes=virtual&&linux/8/console:

```
09:59:18 FAILURE 0.10s J1 | ClusterApplierServiceTests.testClusterStateUpdateLogging <<< FAILURES!
09:59:18    > Throwable #1: java.lang.AssertionError: 
09:59:18    > Expected: an empty collection
09:59:18    >      but: <[Attempted to append to non-started appender mock]>
09:59:18    > 	at __randomizedtesting.SeedInfo.seed([D88B37E396481A0F:717DA27CD857F7D3]:0)
09:59:18    > 	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
09:59:18    > 	at org.elasticsearch.test.ESTestCase.checkStaticState(ESTestCase.java:462)
09:59:18    > 	at org.elasticsearch.test.ESTestCase.after(ESTestCase.java:341)
09:59:18    > 	at java.lang.Thread.run(Thread.java:748)
```

--- a/server/src/test/java/org/elasticsearch/cluster/service/MasterServiceTests.java
+++ b/server/src/test/java/org/elasticsearch/cluster/service/MasterServiceTests.java
@@ -328,11 +328,16 @@ public class MasterServiceTests extends ESTestCase {
                 MasterService.class.getCanonicalName(),
                 Level.DEBUG,
                 "*processing [test3]: took [3s] done publishing updated cluster state (version: *, uuid: *)"));
+        mockAppender.addExpectation(
+            new MockLogAppender.SeenEventExpectation(
+                "test4",
+                MasterService.class.getCanonicalName(),
+                Level.DEBUG,
+                "*processing [test4]: took [0s] no change in cluster state"));
 
         Logger clusterLogger = LogManager.getLogger(MasterService.class);
         Loggers.addAppender(clusterLogger, mockAppender);
         try {
-            final CountDownLatch latch = new CountDownLatch(4);
             masterService.currentTimeOverride = System.nanoTime();
             masterService.submitStateUpdateTask("test1", new ClusterStateUpdateTask() {
                 @Override
@@ -342,9 +347,7 @@ public class MasterServiceTests extends ESTestCase {
                 }
 
                 @Override
-                public void clusterStateProcessed(String source, ClusterState oldState, ClusterState newState) {
-                    latch.countDown();
-                }
+                public void clusterStateProcessed(String source, ClusterState oldState, ClusterState newState) { }
 
                 @Override
                 public void onFailure(String source, Exception e) {
@@ -364,9 +367,7 @@ public class MasterServiceTests extends ESTestCase {
                 }
 
                 @Override
-                public void onFailure(String source, Exception e) {
-                    latch.countDown();
-                }
+                public void onFailure(String source, Exception e) { }
             });
             masterService.submitStateUpdateTask("test3", new ClusterStateUpdateTask() {
                 @Override
@@ -376,9 +377,7 @@ public class MasterServiceTests extends ESTestCase {
                 }
 
                 @Override
-                public void clusterStateProcessed(String source, ClusterState oldState, ClusterState newState) {
-                    latch.countDown();
-                }
+                public void clusterStateProcessed(String source, ClusterState oldState, ClusterState newState) { }
 
                 @Override
                 public void onFailure(String source, Exception e) {
@@ -394,21 +393,18 @@ public class MasterServiceTests extends ESTestCase {
                 }
 
                 @Override
-                public void clusterStateProcessed(String source, ClusterState oldState, ClusterState newState) {
-                    latch.countDown();
-                }
+                public void clusterStateProcessed(String source, ClusterState oldState, ClusterState newState) { }
 
                 @Override
                 public void onFailure(String source, Exception e) {
                     fail();
                 }
             });
-            latch.await();
+            assertBusy(mockAppender::assertAllExpectationsMatched);
         } finally {
             Loggers.removeAppender(clusterLogger, mockAppender);
             mockAppender.stop();
         }
-        mockAppender.assertAllExpectationsMatched();
     }
 
     public void testClusterStateBatchedUpdates() throws BrokenBarrierException, InterruptedException {

